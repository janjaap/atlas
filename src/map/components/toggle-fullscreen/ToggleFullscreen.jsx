import React from 'react';
import PropTypes from 'prop-types';

import MinimizeIcon from '../../../../public/images/icon-minimize.svg';
import MaximizeIcon from '../../../../public/images/icon-maximize.svg';

import './_toggle-fullscreen.scss';

const ToggleFullscreen = ({ isFullscreen, onToggleFullscreen }) => (
  <button
    title={isFullscreen ? 'Kaart verkleinen' : 'Kaart vergroten'}
    className="toggle-fullscreen"
    onClick={() => onToggleFullscreen()}
  >
    {isFullscreen ?
      <MinimizeIcon className="toggle-fullscreen__icon" /> :
      <MaximizeIcon className="toggle-fullscreen__icon" />}
  </button>
);

ToggleFullscreen.propTypes = {
  isFullscreen: PropTypes.bool.isRequired,
  onToggleFullscreen: PropTypes.func.isRequired
};

export default ToggleFullscreen;
