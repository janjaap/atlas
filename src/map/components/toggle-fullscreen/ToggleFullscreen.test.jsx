import React from 'react';
import { shallow } from 'enzyme';

import ToggleFullscreen from './ToggleFullscreen';

describe('ToggleFullscreen', () => {
  describe('actions', () => {
    it('should trigger toggle off when clicked', () => {
      const onToggleFullscreen = jest.fn();
      const wrapper = shallow(
        <ToggleFullscreen
          isFullscreen={true}
          onToggleFullscreen={onToggleFullscreen}
        />
      );
      wrapper.find('button').at(0).simulate('click');
      expect(onToggleFullscreen).toHaveBeenCalled();
    });

    it('should trigger toggle on when clicked', () => {
      const onToggleFullscreen = jest.fn();
      const wrapper = shallow(
        <ToggleFullscreen
          isFullscreen={false}
          onToggleFullscreen={onToggleFullscreen}
        />
      );
      wrapper.find('button').at(0).simulate('click');
      expect(onToggleFullscreen).toHaveBeenCalled();
    });

    describe('rendering', () => {
      it('should render with fullscreen is turned on', () => {
        const onToggleFullscreen = jest.fn();
        const wrapper = shallow(
          <ToggleFullscreen
            isFullscreen={true}
            onToggleFullscreen={onToggleFullscreen}
          />
        );
        expect(wrapper).toMatchSnapshot();
      });

      it('should render with fullscreen is turned off', () => {
        const onToggleFullscreen = jest.fn();
        const wrapper = shallow(
          <ToggleFullscreen
            isFullscreen={false}
            onToggleFullscreen={onToggleFullscreen}
          />
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
