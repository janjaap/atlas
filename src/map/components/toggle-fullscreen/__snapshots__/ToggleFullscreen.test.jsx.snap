// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleFullscreen actions rendering should render with fullscreen is turned off 1`] = `
<button
  className="toggle-fullscreen"
  onClick={[Function]}
  title="Kaart vergroten"
>
  <Component
    className="toggle-fullscreen__icon"
  />
</button>
`;

exports[`ToggleFullscreen actions rendering should render with fullscreen is turned on 1`] = `
<button
  className="toggle-fullscreen"
  onClick={[Function]}
  title="Kaart verkleinen"
>
  <Component
    className="toggle-fullscreen__icon"
  />
</button>
`;
